src/charm.py

```diff
@@ -361,38 +330,40 @@ def _get_runner_manager(
                 image="jammy",
                 service_token=self.service_token,
                 lxd_storage_path=lxd_storage_path,
-                charm_state=self.state,
-                dockerhub_mirror=self.state.charm_config.dockerhub_mirror,
+                proxy_config=state.proxy_config,
+                arch=state.arch,
+                ssh_debug_connections=state.ssh_debug_connections,
+                is_metrics_logging_available=state.is_metrics_logging_available,
+                dockerhub_mirror=state.charm_config.dockerhub_mirror,
             ),
         )
 
     @catch_charm_errors
-    @_setup_github_runner_charm_state
     def _on_install(self, _event: InstallEvent) -> None:
         """Handle the installation of charm.
 
         Args:
             event: Event of installing the charm.
         """
-        self.unit.status = MaintenanceStatus("Installing packages")
+        state = self._setup_state()
```

Comment:
As mentioned in the roadmap meeting, letting the install hook go into BlockedStatus (which can happen with an invalid config) may not be desirable, as the install hook will not be retriggered.

This may also apply to start hook.

Reply:
We can also create a bug ticket and resolve it later, as it is not strictly related to this PR.

Reply:
I have created a ticket for this bug.